File: 프로그래머스/1/12932. 자연수 뒤집어 배열로 만들기/자연수 뒤집어 배열로 만들기.py
def solution(n):
    answer = []
    num1 = 10000000000
    
    for i in range(11):
        
        insert_num = n // num1
        
        if insert_num != 0 or answer != []:
            answer.insert(0,insert_num)
        
        n -= num1*insert_num
        num1 /= 10
        
        
    return answer

File: 프로그래머스/1/12932. 자연수 뒤집어 배열로 만들기/자연수 뒤집어 배열로 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(long long n) {
    // 리턴할 값은 메모리를 동적 할당해주세요.
    long long mul = 10000000000;
    int temp = 0;
    int* answer = (int*)malloc(sizeof(int)*11);
    for(int i = 10; i >=0 ; i--){
        answer[i] = n / mul;
        n -= answer[i] *mul;
        mul /= 10;
    }
    return answer;
}